src/components/drawer/drawer.component.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+
+import {
+  StyledDrawerWrapper,
+  StyledDrawerContent,
+  StyledIconButton,
+  StyledDrawerChildren,
+  StyledDrawerSidebar
+} from './drawer.style';
+
+const Drawer = ({
+  defaultExpanded,
+  expanded,
+  onChange,
+  children,
+  expandedWidth,
+  sidebar,
+  animationSpeed,
+  ...props
+}) => {
+  const isControlled = expanded !== undefined;
+  const [isExpandedInternal, setIsExpandedInternal] = useState(defaultExpanded || false);
+  const [isFirstOpen, setFirstOpen] = useState(true);
+  const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const toggleDrawer = useCallback((ev) => {
+    if (!isControlled) setIsExpandedInternal(!isExpanded);
+    if (onChange) onChange(ev, !isExpanded);
+    setFirstOpen(false);
+  }, [isControlled, isExpanded, onChange]);
+
+  const closedClass = isFirstOpen ? '' : 'closed';
+
+  return (
+    <StyledDrawerWrapper
+      data-element='drawer'
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedWidth }
+        animationSpeed={ animationSpeed }
+        className={ isExpanded ? 'open' : closedClass }
+      >
+        <StyledIconButton
+          data-element='drawer-toggle'
```

Comment:
I think we should look to introduce some `aria` roles here to indicate that the button controls the menu.

![image](https://user-images.githubusercontent.com/2328042/79328191-8de32f00-7f0d-11ea-8b96-31f0b4d38e4c.png)

![image](https://user-images.githubusercontent.com/2328042/79328238-9dfb0e80-7f0d-11ea-9079-6f2badda937a.png)

Details here https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

Reply:
good call, added aria-expanded

Reply:
added aria-label and role.